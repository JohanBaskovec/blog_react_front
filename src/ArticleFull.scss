@import "variables";

.ArticleFull {
  display: flex;
  flex-direction: column;
  &__author_name {
    color: $grey3;
    margin-bottom: 1rem;
  }

  &__title {
    margin-bottom: 0.5rem;
  }
}
