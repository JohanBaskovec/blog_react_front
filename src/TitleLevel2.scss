.TitleLevel2 {
  font-size: 1.4rem;
  font-weight: bold;
  margin: 0;
}
