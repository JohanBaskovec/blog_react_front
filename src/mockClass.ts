

type constructorTypeOf<T> = new (...args:any[]) => T;

export function mockDefaultApi<T>(o: T): T {
    const mocked: any = {};
    for (const key in o) {
        if (typeof o[key] === 'function') {
            mocked[key] = jest.fn();
        }
    }
    return mocked;
}

type MockedProxy<T> = jest.Mock<T extends (...args: any[]) => infer U ? U : T, any[]>;

type MockedClass<T> = {
    [k in keyof T]: MockedProxy<T[k]>;
}

export function mockClass<T>(constr: constructorTypeOf<T>): T & MockedClass<T>{
    let obj: T = new constr();
    const mocked: any = {};

    do {
        const properties = Object.getOwnPropertyNames(obj);
        for (const propertyName of properties) {
            if (typeof (obj as any)[propertyName] === "function" && !mocked[propertyName]) {
                mocked[propertyName] = jest.fn();
            }
        }
    } while (obj = Object.getPrototypeOf(obj));
    return mocked as any;
}
