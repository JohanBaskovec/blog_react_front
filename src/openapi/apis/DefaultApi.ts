// tslint:disable
/**
 * Blog
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 0.0.0
 * Contact: johan@johanbaskovec.fr
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { Observable } from 'rxjs';
import { BaseAPI, HttpHeaders, throwIfNullOrUndefined, encodeURI } from '../runtime';
import {
    Article,
    ArticleFormData,
    LoginForm,
    RegistrationForm,
    User,
} from '../models';

export interface GetArticleByIdRequest {
    id: string;
}

export interface InsertArticleRequest {
    articleFormData: ArticleFormData;
}

export interface LoginRequest {
    loginForm: LoginForm;
}

export interface RegisterRequest {
    registrationForm: RegistrationForm;
}

export interface UpdateArticleRequest {
    articleFormData: ArticleFormData;
}

/**
 * no description
 */
export class DefaultApi extends BaseAPI {

    /**
     * Get all the articles
     */
    getAllArticles = (): Observable<Array<Article>> => {
        return this.request<Array<Article>>({
            path: '/article',
            method: 'GET',
        });
    };

    /**
     * Get an article
     */
    getArticleById = ({ id }: GetArticleByIdRequest): Observable<Article> => {
        throwIfNullOrUndefined(id, 'getArticleById');

        return this.request<Article>({
            path: '/article/{id}'.replace('{id}', encodeURI(id)),
            method: 'GET',
        });
    };

    /**
     * Get the current session
     */
    getCurrentAuthenticatedUser = (): Observable<User> => {
        const headers: HttpHeaders = {
        };

        return this.request<User>({
            path: '/me',
            method: 'GET',
            headers,
        });
    };

    /**
     * Insert an article
     */
    insertArticle = ({ articleFormData }: InsertArticleRequest): Observable<void> => {
        throwIfNullOrUndefined(articleFormData, 'insertArticle');

        const headers: HttpHeaders = {
            'Content-Type': 'application/json',
        };

        return this.request<void>({
            path: '/article',
            method: 'POST',
            headers,
            body: articleFormData,
        });
    };

    /**
     * Create a new HTTP session
     */
    login = ({ loginForm }: LoginRequest): Observable<User> => {
        throwIfNullOrUndefined(loginForm, 'login');

        const headers: HttpHeaders = {
            'Content-Type': 'application/json',
        };

        return this.request<User>({
            path: '/login',
            method: 'POST',
            headers,
            body: loginForm,
        });
    };

    /**
     * Logout
     */
    logout = (): Observable<void> => {
        const headers: HttpHeaders = {
        };

        return this.request<void>({
            path: '/logout',
            method: 'POST',
            headers,
        });
    };

    /**
     * Register
     */
    register = ({ registrationForm }: RegisterRequest): Observable<void> => {
        throwIfNullOrUndefined(registrationForm, 'register');

        const headers: HttpHeaders = {
            'Content-Type': 'application/json',
        };

        return this.request<void>({
            path: '/register',
            method: 'POST',
            headers,
            body: registrationForm,
        });
    };

    /**
     * Update an article
     */
    updateArticle = ({ articleFormData }: UpdateArticleRequest): Observable<void> => {
        throwIfNullOrUndefined(articleFormData, 'updateArticle');

        const headers: HttpHeaders = {
            'Content-Type': 'application/json',
        };

        return this.request<void>({
            path: '/article',
            method: 'PUT',
            headers,
            body: articleFormData,
        });
    };

}
