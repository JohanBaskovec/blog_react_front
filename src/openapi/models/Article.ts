// tslint:disable
/**
 * Blog
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 0.0.0
 * Contact: johan@johanbaskovec.fr
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

/**
 * @export
 * @interface Article
 */
export interface Article {
    /**
     * @type {string}
     * @memberof Article
     */
    id: string;
    /**
     * @type {string}
     * @memberof Article
     */
    title: string;
    /**
     * @type {string}
     * @memberof Article
     */
    content: string;
}
