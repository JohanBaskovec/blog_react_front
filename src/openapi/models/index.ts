export * from './Article';
export * from './HttpSession';
export * from './LoginForm';
export * from './RegistrationForm';
export * from './User';
