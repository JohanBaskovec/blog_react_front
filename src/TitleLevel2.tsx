import "./TitleLevel2.scss";
import React from "react";

interface TitleLevel2Props {
    children?: React.ReactNode;
    style?: React.CSSProperties;
}

export function TitleLevel2(props: TitleLevel2Props) {
    return <h2 className="TitleLevel2" style={props.style}>{props.children}</h2>
}
