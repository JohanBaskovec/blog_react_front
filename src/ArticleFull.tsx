import {DefaultApi} from "./openapi/apis";
import {Article} from "./openapi/models";
import React, {useEffect, useState} from "react";
import {useParams} from "react-router-dom";
import {ApiError, ApiErrorType} from "./ApiError";
import {TitleLevel2} from "./TitleLevel2";
import {AppLink} from "./AppLink";

export interface ArticleFullProps {
    api: DefaultApi;
}

export interface ArticleFullPathParams {
    id: string;
}

export function ArticleFull({api}: ArticleFullProps) {
    const {id}: ArticleFullPathParams = useParams<ArticleFullPathParams>();
    const [article, setArticle] = useState<Article | null>(null);
    const [error, setError] = useState<ApiError | null>(null);
    useEffect(() => {
        api.getArticleById({id: id}).subscribe((article) => {
                setArticle(article);
            },
            (error) => {
                setError(ApiError.fromError(error));
            });
    }, [])

    if (error) {
        if (error.type === ApiErrorType.notFound) {
            return <div>This article doesn't exist.</div>
        } else {
            return <div>An error happened, please try again later.</div>
        }
    }
    return (
        <article className="ArticleFull">
            <TitleLevel2 style={{marginBottom: "1rem"}}>
                {article?.title}
            </TitleLevel2>
            <div className="ArticleFull__editionLink">
                <AppLink to={'/article/' + article?.id + '/edit'}>Edit</AppLink>
            </div>
            <div className="ArticleFull__content">
                {article?.content}
            </div>
        </article>
    )
}
