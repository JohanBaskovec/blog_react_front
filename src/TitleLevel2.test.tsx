import React from "react";
import {DefaultApi, GetArticleByIdRequest} from "./openapi/apis";
import {of, throwError} from "rxjs";
import {mockDefaultApi} from "./mockClass";
import {render, unmountComponentAtNode} from "react-dom";
import {act} from "react-dom/test-utils";
import pretty from "pretty";
import {Article} from "./openapi/models";
import {AjaxError} from "rxjs/ajax";
import {ArticleFull} from "./ArticleFull";
import {MemoryRouter} from "react-router-dom";
import {TitleLevel2} from "./TitleLevel2";

let container: HTMLDivElement;
beforeEach(() => {
    // setup a DOM element as a render target
    container = document.createElement("div");
    document.body.appendChild(container);
});

afterEach(() => {
    // cleanup on exiting
    if (container != null) {
        unmountComponentAtNode(container);
        container.remove();
    }
});

describe('TitleLevel2', () => {
    it("renders properly", () => {
        act(() => {
            render( <TitleLevel2>Title</TitleLevel2> , container);
        });
        expect(pretty(container.innerHTML)).toMatchSnapshot();
    });
});
