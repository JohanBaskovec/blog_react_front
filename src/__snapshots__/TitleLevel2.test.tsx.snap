// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TitleLevel2 renders properly 1`] = `"<h2 class=\\"TitleLevel2\\">Title</h2>"`;
